lib/fizzy/execute.hpp

```diff
@@ -2,30 +2,35 @@
 
 #include "types.hpp"
 #include <cstdint>
+#include <functional>
 
 namespace fizzy
 {
+// The result of an execution.
+struct execution_result
+{
+    // true if execution resulted in a trap
+    bool trapped;
+    // the resulting stack (e.g. return values)
+    // NOTE: this can be either 0 or 1 items
+    std::vector<uint64_t> stack;
+};
+
+using ImportedFunction = execution_result (*)(const std::vector<uint64_t>&);
```

Comment:
Should pass the `instance`, because many imported functions likely need access to the memory.